e2e/runner/accounting.go

```diff
@@ -150,95 +143,143 @@ func (r *E2ERunner) CheckSolanaTSSBalance() error {
 	gatewayBalance, err := r.SuiGetGatewaySUIBalance()
-	if err != nil {
-		return fmt.Errorf("failed to get SUI balance for Sui gateway: %w", err)
-	}
+	require.NoError(r, err, "failed to get SUI balance for Sui gateway: %w", err)
 
 	zrc20Supply, err := r.SUIZRC20.TotalSupply(&bind.CallOpts{})
-	if err != nil {
-		return err
-	}
+	require.NoError(r, err)
 
 	// subtract value from the gas stability pool because of the artificial minting bug
 	// TODO: remove on the chain upgrade to v33
 	// https://github.com/zeta-chain/node/issues/4034
 	gasStabiltiyPoolBalance, err := r.SUIZRC20.BalanceOf(&bind.CallOpts{}, fungibletypes.GasStabilityPoolAddressEVM())
-	if err != nil {
-		return fmt.Errorf("failed to get SUI gas stability pool balance: %w", err)
-	}
+	require.NoError(r, err, "failed to get SUI gas stability pool balance: %w", err)
 	zrc20Supply = zrc20Supply.Sub(zrc20Supply, gasStabiltiyPoolBalance)
 
 	// Subtract 0.1 SUI to take in consideration the 0.1 SUI minted in the gas pool
 	// TODO: a proper implementation is to implement a donate method in Sui Contract and use it to donate 0.1 SUI
 	// https://github.com/zeta-chain/protocol-contracts-sui/issues/58
 	zrc20Supply = zrc20Supply.Sub(zrc20Supply, big.NewInt(100000000))
 
-	if gatewayBalance.Cmp(zrc20Supply) < 0 {
-		return fmt.Errorf("SUI: TSS balance (%d) < ZRC20 TotalSupply (%d) ", gatewayBalance, zrc20Supply)
-	}
+	require.GreaterOrEqual(
+		r,
+		gatewayBalance.Cmp(zrc20Supply),
+		0,
+		"SUI: TSS balance (%d) < ZRC20 TotalSupply (%d) ",
+		gatewayBalance,
+		zrc20Supply,
+	)
 	r.Logger.Info("SUI: TSS balance (%d) >= ZRC20 TotalSupply (%d)", gatewayBalance, zrc20Supply)
-	return nil
 }
 
-func (r *E2ERunner) checkERC20TSSBalance() error {
+func (r *E2ERunner) checkERC20TSSBalance() {
 	custodyBalance, err := r.ERC20.BalanceOf(&bind.CallOpts{}, r.ERC20CustodyAddr)
-	if err != nil {
-		return err
-	}
+	require.NoError(r, err)
 
 	erc20zrc20Supply, err := r.ERC20ZRC20.TotalSupply(&bind.CallOpts{})
-	if err != nil {
-		return err
-	}
-	if custodyBalance.Cmp(erc20zrc20Supply) < 0 {
-		return fmt.Errorf("ERC20: custody balance (%d) < ZRC20 TotalSupply (%d) ", custodyBalance, erc20zrc20Supply)
-	}
+	require.NoError(r, err)
+
+	require.GreaterOrEqual(
+		r,
+		custodyBalance.Cmp(erc20zrc20Supply),
+		0,
+		"ERC20: custody balance (%d) < ZRC20 TotalSupply (%d) ",
+		custodyBalance,
+		erc20zrc20Supply,
+	)
 	r.Logger.Info("ERC20: TSS balance (%d) >= ERC20 ZRC20 TotalSupply (%d)", custodyBalance, erc20zrc20Supply)
-	return nil
 }
 
-func (r *E2ERunner) checkZetaTSSBalance() {
-	// get total ZETA locked
+func (r *E2ERunner) checkZetaTSSBalance(testLegacy bool) {
+	// zetaMintedPoolSetup is the amount of Zeta minted to add liquidy to a gas token pool when setting it up
+	// https://github.com/zeta-chain/node/blob/854040f044d198a0453a7b9245d544debd9da055/x/fungible/keeper/gas_coin_and_pool.go#L88
+	zetaMintedPoolSetup, ok := sdkmath.NewIntFromString("400000000000000000")
```

Comment:
Does `0.4ZETA` comes from deployment of zrc20 `ETH, BTC, SOL, SUI, TON`?

Reply:
It's strange, we also have `SUI` deployed [here](https://github.com/zeta-chain/node/blob/develop/e2e/runner/setup_sui.go#L233), totally 0.5

Reply:
Yeah , thanks for catching that , f90f873d789f4574632f8f2b9abdab66a2530736